From d9d14857b2eec9c1fb9f2bd12850eacad9980c99 Mon Sep 17 00:00:00 2001
From: Anya Lin <hukk1996@gmail.com>
Date: Tue, 30 Jul 2024 13:04:04 +0100
Subject: [PATCH 3/3] allow specified address

---
 main.go | 26 ++++++++++++++++++++++----
 1 file changed, 22 insertions(+), 4 deletions(-)

diff --git a/main.go b/main.go
index 266365a..0ef4cc3 100644
--- a/main.go
+++ b/main.go
@@ -90,11 +90,29 @@ func main() {
 	loggingHandler := NewApacheLoggingHandler(mux, os.Stdout) // HTTP access log is sent to stdout for now
 
 	// HTTP server setup
-	plainHttpServer := &http.Server{
-		Addr:    ":" + strconv.Itoa(localResolvePort),
-		Handler: loggingHandler,
+	if !(localResolveIp4Enabled || localResolveIp6Enabled) {
+		plainHttpServer := &http.Server{
+			Addr:    ":" + strconv.Itoa(localResolvePort),
+			Handler: loggingHandler,
+		}
+		go plainHttpServer.ListenAndServe()
+	}
+
+	if localResolveIp4Enabled {
+		plainHttpServer4 := &http.Server{
+			Addr:    localResolveIp4Address.String() + ":" + strconv.Itoa(localResolvePort),
+			Handler: loggingHandler,
+		}
+		go plainHttpServer4.ListenAndServe()
+	}
+
+	if localResolveIp6Enabled {
+		plainHttpServer6 := &http.Server{
+			Addr:    "[" + localResolveIp6Address.String() + "]:" + strconv.Itoa(localResolvePort),
+			Handler: loggingHandler,
+		}
+		go plainHttpServer6.ListenAndServe()
 	}
-	go plainHttpServer.ListenAndServe()
 
 	// HTTPS server setup
 	// tlsHttpServer := &http.Server{
-- 
2.45.2

